dipy/denoise/localpca.py

```diff
@@ -40,14 +41,22 @@ def _pca_classifier(L, nvoxels):
            theory. Neuroimage 142:394-406.
            doi: 10.1016/j.neuroimage.2016.08.016
     """
+
+    # if num_samples - 1 (to correct for mean subtraction) is less than number
+    # of features, discard the zero eigenvalues
+    if L.size > nvoxels - 1:
+        L = L[-(nvoxels - 1):] 
+
     var = np.mean(L)
     c = L.size - 1
-    r = L[c] - L[0] - 4 * np.sqrt((c + 1.0) / nvoxels) * var
+    # NOTE: data is centered, at most (nvoxels - 1) non-zero eigenvalues
+    r = L[c] - L[0] - 4 * np.sqrt((c + 1.0) / (nvoxels - 1)) * var
     while r > 0:
         var = np.mean(L[:c])
         c = c - 1
-        r = L[c] - L[0] - 4 * np.sqrt((c + 1.0) / nvoxels) * var
+        r = L[c] - L[0] - 4 * np.sqrt((c + 1.0) / (nvoxels - 1)) * var
```

Comment:
Following https://github.com/dipy/dipy/discussions/2678#discussioncomment-4336266, https://github.com/dipy/dipy/discussions/2678#discussioncomment-4342309 on the number of directions vs. voxels, and Rafael's comment
>  You can easily rewrite the equation in Veraart et al., 2016 to match the code in function _pca_classifier.

maybe such re-arrangements could be explicitly put allowing to identify the equations. Otherwise, without writing those oneself, looks difficult to review. It also requires to go through the issue comments, which is unlikely to happen if one simply opens the DIPY documentation.

Reply:
Either in the original form or the re-arranged form, but documenting either form (method docstring and before each relevant block) would help users and reviewers. I find it helpful writing those using LaTeX syntax.

Reply:
I have added documentation to explain that this while loop, and the condition therein, is expressed differently to the paper, and I have identified the corresponding equation numbers so people can look themselves.

Reply:
OK. I'd have preferred to write the equations explicitly in the documentation using LaTeX, but OK. Resolving.